pilot/api/data.py

```diff
@@ -484,12 +591,49 @@ def resolve_surl(self, fspec, protocol, ddmconf, **kwargs):
             raise PilotException('Failed to resolve ddmendpoint by name=%s' % fspec.ddmendpoint)
 
         if not ddm.is_deterministic:
-            raise PilotException('resolve_surl(): Failed to construct SURL for non deterministic ddm=%s: NOT IMPLEMENTED', fspec.ddmendpoint)
-
-        surl = protocol.get('endpoint', '') + os.path.join(protocol.get('path', ''), self.get_path(fspec.scope, fspec.lfn))
+            if ddm.type in ['OS_ES', 'OS_LOGS']:
+                surl = self.resolve_surl_os(fspec, protocol, ddm)
+            else:
+                raise PilotException('resolve_surl(): Failed to construct SURL for non deterministic ddm=%s: NOT IMPLEMENTED', fspec.ddmendpoint)
+        else:
+            surl = protocol.get('endpoint', '') + os.path.join(protocol.get('path', ''), self.get_path(fspec.scope, fspec.lfn))
 
         return {'surl': surl}
 
+    @classmethod
```

Comment:
this function was removed in one of the last pull requests and should now be imported from the new checksum module. did you sync your branch before the pull request? it might be that some recent changes will be overwritten by your pull request.